### DISCUSSION TITLE
Behavior of the cache in case of client with their own upstream DNS

### DISCUSSION THREAD

**Discussion by laurentftech** [AdguardTeam/AdGuardHome#4226] [General]
Hi. 
It seems that if a client is not using the general upstream DNS, he is not using the Adguard Home cache either. Is it the intended behavior? In that case, it should be documented. 
Thanks.

**Comment by ainar-g** [comment]
Hello.  Could you please elaborate?  What settings and options are enabled for the client in question?

**Reply by laurentftech** [reply]
I noticed that if I set a certain upstream dns (https://doh.CleanBrowsing.org/doh/adult-filter for exemple) for a specific client, he is not using the cache. 
If I remove the upstream dns for this client, he will used the general upstream dns and will use the cache. 
It sounds right to me (it could be an option at client level though) but I was wondering why some of my clients were using the cache and some not.

**Reply by ainar-g** [reply]
@Gandulf78, I'll look into it deeper next week.  Thanks for the report!

**Reply by ainar-g** [reply]
@Gandulf78, see AdguardTeam/dnsproxy#169.  We'll probably try to approach that in v0.108.

**Reply by laurentftech** [reply]
Yes, that's the same. Thank you.
